fleetspeak/src/client/config/config.go

```diff
@@ -92,6 +73,27 @@ type Configuration struct {
 
 	// RevokedCertSerials is a list of certificate serial numbers which have been
 	// revoked. Revoked serial numbers can also be provided by the server and will
-	// stored to the writeback file, when Ephemeral is not set.
+	// stored to the writeback location, if NoopPersistenceHandler is not used.
 	RevokedCertSerials [][]byte
 }
+
+const (
+	communicatorFilename  = "communicator.txt"
+	signedServicesDirname = "services"
+	writebackFilename     = "writeback"
+)
+
+// PersistenceHandler manages client's configuration storage.
+type PersistenceHandler interface {
+	ReadState() (*clpb.ClientState, error)
+	WriteState(*clpb.ClientState) error
+	ReadCommunicatorConfig() (*clpb.CommunicatorConfig, error)
+
+	ReadSignedServices() ([]*fspb.SignedClientServiceConfig, error)
+	SaveSignedService(*fspb.SignedClientServiceConfig) error
+}
+
+// NewDefaultPersistenceHandler returns the default persistence handler for the current platform.
+func NewDefaultPersistenceHandler(configurationPath string, readonly bool) (PersistenceHandler, error) {
```

Comment:
How much does the layering of indirection (entrypoint -> NewDefault -> Plaform NewDefault -> Implementation) buy us? The windows client will need its own entry point to handle the service interfacing stuff, right?

If we have a windows-specific main function, it could plug in the windows specific persistence handler. Then we lose this method, the two implementations of it, and it becomes even more obvious what implementation we'll end up with when looking at the configuration setup.

It also would allow additional control in the future, e.g. some managers could take additional parameters, etc.

Reply:
Discussed offline.

Reply:
Clarified as much as possible in the doccomment. Hopefully no one will get confused here